### ISSUE TITLE
looking for a RF frontend for our OVI40 SDR project

### ISSUE THREAD

**Issue by df8oe** [xaxaxa/vhdl#1]
Hi,
I am looking for a RF frontend for our OVI40 amateur radio project which is a standalone TRX based on STM32 - firmware is called UHSDR. Actually I am thinking of usage of Pluto SDR and I found your "bootleg Pluto" in eevblog. It looks very nice and I am wondering if you do have more informations about your project. Is it still alive? I want to do data transfer via SPI (or I2S) - there are not very large bandwidths needed. If we can get 1MHz we are very happy because of RAM of STM32 is not so big... SDR4 seems to be homebrewable...

**Comment by gabriel-tenma-white** [comment]
The successor to that (sdr5) is posted here: https://github.com/gabriel-tenma-white/sdr5
sdr5 hasn't been prototyped and verified yet, but should be in a month or so. 

I can also upload the sdr4 design files. There are exactly two errata which are (1) RESETB pin of the AD9363 should be wired to an IO pin on the FPGA and (2) the 1.8V rail buck converter should be replaced with a linear regulator, current draw on that rail is very low (<200mA) and all available SOT26 buck ICs implement pulse-skipping at low current output which adds extreme amounts of noise to the rail. The two errata are bodge fixed on the prototype unit but we aren't planning to make another revision of sdr4 because it is superseded by sdr5 going forward.

**Comment by gabriel-tenma-white** [comment]
If this is for HF only I would recommend a direct digitization design, since it's possible to get 10 bit ADCs of up to 80Msps for fairly cheap. The ADC10080 goes for about $1-$2. You will need a pre-filter on the input anyway, so you can also do sub-sampling to cover >40MHz. I do have a ADC10080 to XC6SLX9 design that can be modified to add RF frontend and filter bank. For transmit I would use a ADV7123.

**Comment by df8oe** [comment]
Many thanks for your fast reply. sdr5 looks very interesting. It will be capable to carry most stuff Pavel Denim has published for RedPitaya. It will close the issue of Pluto that flash size is very low.
OVI40 shall become a portable SDR device which can be used to work via Oscar 100. So it is not only a RF TRYX but must be capable to work much higher bands, too. I have worked with N2ADR ideas, too. Direct sampling I would do using AD9866 which has 12 bit. But I am unsure and in doubt what is the right decision. I must balance cost, performance and homebrewing possibility. And there will be not ONE concept of best proof. There will be many all dependent on sight of view. Actually I am experimenting with AD9363 and upconverter for RF bands. Performance is absolutely acceptable, signal quality in TX is very good.

SDR4 and SDR5 are better from specific views: SDR4 is cheaper and can be homebrewed (not only for me but for the mass of amateur radio operators) and SDR5 combines Red Pitaya horse power with AD9363. Maybe there would be possible to add direct conversion for lower bands because there are enough free gpios on Zync. I think I will build a SDR4 (with improved design as you told) and check out performance...

**Comment by gabriel-tenma-white** [comment]
I would say that a cheap ADC and direct sampling will get you far better overall system dynamic range than doing upconversion before the AD9363, since the AD9363 already has shit linearity (-18dBm IIP3). I would like to substitute the LMS7002M for the AD9363 but unfortunately the Lime part is impossible to source and comes in a stupid package and pin assignment that is probably impossible to route on a 4 layer board. 

For reference these are the current spot prices and relative sourcing difficulties for the parts mentioned:
- AD9363 - $10 - medium
- AD9866 - $10 - medium-hard
- ADC10080 - $1.6 - medium
- SDA7123 (ADV7123 substitute) - $1 - easy
- XC6SLX9 - $4 - easy
- XC7Z010-1CLG400C - $13 - medium-easy

I still need to do more research on ADC part selection since I'm still looking for a well priced 12 bit ADC at >20Msps and need a second source for ADC10080. The AD9866 is also difficult to source and I need to research alternatives.

**Comment by df8oe** [comment]
Yes, compromises regarding costs/availability/performance are not easy... Can I find your ADC10080 to XC6SLX9 design somewhere in your repos?

**Comment by df8oe** [comment]
What about AFE7225? It is not so cheap but it offers high performance and combines TX and RX  as AD9866...

**Comment by gabriel-tenma-white** [comment]
I'm going to be uploading those to a repo; will comment back when I do.

AFE7225/7222 doesn't seem to be obtainable; it looks like probably separate ADC/DAC is going to be easier. From 10 bits to 12 bits the prices jump way up, so I would look into whether 12 bit is really necessary. I would probably start with a cheap FPGA+ADC/DAC+amplifier+filters board, evaluate performance, and see where the bottleneck is. If you are only going to be receiving a narrowband channel at a time then some DSP can go a long way. For example if channel bandwidth is 40kHz, sample rate is 80Msps, ADC bits is 10, then effective bits after process gain is 15 bits.

**Comment by df8oe** [comment]
You are absolutely right. I recognized this few hours before - now you are pointing me the same way. I will start using a cheap 10 bit ADC as ADC10080. Thank you for all your time pointing me up this direction.

**Comment by df8oe** [comment]
I have found some zips in eevblog regarding sdr4. There are PCBs and some Linux testing tools for it (very nice - I am working under Linux only...). PCBs are in gEDA format - very nice, too. My favourite EDA is KiCAD but of course I do have gEDA installed, too. What I do not have found are source files for FPGA. For my task FPGA must only decimate data stream to an I2S similar serial interface (acting as I2S master) at some specific sample rates (e.g. 192k, 384K, 768k and 1536K). It can receive commands via SPI or I2C. Spartan you use has less logic cells as Cyclone N2ADR uses but it has less work to do so I guess it is possible. Because of this FPGA is very cheap frequency span can be divided in two sections (and PCBs): 0...70MHz using direct conversion (40-70MHz sub-sampling) and 70MHz...6GHz using AD9363. So there will be two RF chains which both are not tooooo expensive and firmware for FPGA has wide parts identical.... This looks like a very interesting concept and is homebrewable with easy/medium skills (RF) and medium/hard skills (VHF/SHF) for home brewing. Our OVI40 will be an Open Source (software and hardware) project under GPLv3. So if you can think about your RF data accuisition boards can be part of it I would be happy... I never have had coded using any HDL language. But I am willing to learn much and I have fun in learning :) . Our firmware for UI is written in C (https://github.com/df8oe/UHSDR) and my workstation runs using Arch Linux (I do not have machines with any commercial OS)...

**Comment by gabriel-tenma-white** [comment]
I've uploaded the sdr4 design files and FPGA firmware here: https://github.com/gabriel-tenma-white/sdr4

**Comment by df8oe** [comment]
Hi,
many thanks for doing this. I enjoy our discussion. I will prototype a PCB and implement the corrections for the two mentioned erratas. Then I will test with firmware "unvhanged" and then change step by step (because I must learn working with vdl) the data format, output to I2S like connections instead of USB and adding a SPI which is directly routed to AD9363. How do you build firmware? Do you use ISE or command line?

If you are interested I can send you a bare PCB for OVI40 UI (for free of course).

**Comment by gabriel-tenma-white** [comment]
For building the FPGA logic usually you would use ISE, but I think the tools can be invoked on the command line too if you need to automate it. 

I looked around and haven't been able to find schematics/PCB files of OVI40; since shipping from EU to here (China) is probably expensive I think it makes more sense for me to get it fabbed (with PCBA) in my next panel run. Can you point me to the PCB design files?

**Comment by df8oe** [comment]
This discussion gets now very far away from the github repo it takes place - so we should better discuss personally.... Pse contact me via email: df8oe@gmx.de

**Comment by gabriel-tenma-white** [comment]
Is there a OVI40 repository? I can start an issue there or the UHSDR repo. I don't check my email often but do you have Discord? My invite link is https://discord.gg/4Zrr67
